lib/signals.c

```diff
@@ -181,10 +181,8 @@ static void
 signal_handler(int sig)
 {
 	if (write(signal_pipe[1], &sig, sizeof(int)) != sizeof(int)) {
-		DBG("signal_pipe write error %s", strerror(errno));
 		assert(0);
-
-		log_message(LOG_INFO, "BUG - write to signal_pipe[1] error %s - please report", strerror(errno));
+		fprintf(stderr, "BUG - write to signal_pipe[1] error %s - please report\n", strerror(errno));
```

Comment:
We can't use fprintf, since stdout and stderr are not open (or are opened on /dev/null), and keepalived, unless specifically run in foreground, has no controlling terminal to which it could write.